protocol/consistencychecks.go

```diff
@@ -149,11 +147,26 @@ func (cc *ConsistencyChecks) verifyDirectoryProof(df *DirectoryProof,
 		return
 	}
 
-	if verifResult = cc.verifyReturnedPromise(tb, uname, key, str, ap); verifResult != nil {
-		return
+	// validates the (response code, proof type) according to the request type.
+	switch requestType {
+	case RegistrationType:
+		if responseCode != ErrorNameExisted &&
+			(responseCode != Success || proofType != proofOfAbsence) {
```

Comment:
I should have a comment for this, sorry. It's just the check in `cc.Verify` before.